File: backend/Models/Album.cs
using Models.Base;

namespace Models
{
    public class Album : BaseEntity
    {
        public string Title { get; set; }
        public string Genre { get; set; }
        public string ArtistId { get; set; }
        public Artist Artist { get; set; }
        public List<Track> Tracks { get; set; }
    }
}

File: Frontend/client_app/lib/models/album.dart
import 'package:frontend/models/artist.dart';
import 'package:frontend/models/base/base_entity.dart';
import 'package:frontend/models/track.dart';
import 'package:json_annotation/json_annotation.dart';

part 'album.g.dart';

@JsonSerializable()
class Album extends BaseEntity{
  String title;
  int artistId;
  Artist? artist;
  List<Track> tracks;
  String? coverImageUrl;

  Album({
    required super.id,
    required super.createdAt,
    required super.updatedAt,
    required this.title,
    required this.artistId,
    required this.artist,
    required this.tracks,
    this.coverImageUrl,
  });

  factory Album.fromJson(Map<String, dynamic> json) => _$AlbumFromJson(json);

  Map<String, dynamic> toJson() => _$AlbumToJson(this);
}
